libbeat/esleg/eslegclient/bulkapi_mock_test.go

```diff
@@ -146,3 +149,57 @@ func TestOneHost503Resp_Bulk(t *testing.T) {
 		t.Errorf("Should return <503 Service Unavailable> instead of %v", err)
 	}
 }
+
+func TestEnforceParameters(t *testing.T) {
+	client, _ := NewConnection(ConnectionSettings{
+		Parameters: map[string]string{"hello": "world"},
+		URL:        "http://localhost",
+		Timeout:    0,
+	})
+	client.Encoder = NewJSONEncoder(nil, false)
+
+	client.HTTP = &reqInspector{
+		assert: func(req *http.Request) (*http.Response, error) {
+			assert.Equal(t, "world", req.URL.Query().Get("hello"))
```

Comment:
The change modifies the merging of parameters. We should check if all parameters (preconfigured and local ones) are present.